tests/integration/kubernetes/lib.sh

```diff
@@ -94,11 +94,41 @@ assert_pod_fail() {
 	! k8s_create_pod "$container_config" || /bin/false
 }
 
+
+# Check the pulled rootfs on host for given node and sandbox_id
+#
+# Parameters:
+#	$1 - the k8s worker node name
+#	$2 - the sandbox id for kata container
+#	$3 - the expected count of pulled rootfs
+#
+assert_rootfs_count() {
```

Comment:
Hey @ChengyuZhu6 - looking at the test runs for this (I just picked TDX at random: https://github.com/kata-containers/kata-containers/actions/runs/8047127494/job/21977181269?pr=8484) it looks like the failures aren't the expected ones with 1 vs 2 rootfs directories, but all of them have:
```
count of container rootfs in host is: 0, expect count is: <expected>
```
I guess there are a couple of things that are potentially happening:
- We are no longer pulling the pause rootfs on the host. I don't think we ever fully understood why we were in `CCv0`, but IIRC correctly we still saw this behaviour in the nydus-snapshotter tests *and* there are no longer any issues with the annotation being cached
- Something else is going wrong, so that we are looking for the host side rootfs in the wrong place. I think Choi say something similar to this in the past, in https://github.com/kata-containers/tests/issues/5781, though I think you helped resolve this by suggesting removing a cached kata-shim on that system?

I need to get my own test environment set up soon, once we have support for non-TEE image pull, but I wondered if you've been able to see either of these behaviours, or if you could `kata-runtime exec` into the pod and see if it's been pulled to `/run/image/<cid>` to check it was pulled on the host as expected? I t might even be worth adding that check to the test cases too?

I hope this mades sense and sorry if I've missed something obvious

Reply:
Thanks @stevenhorsman. And I found that the agent failed to build on s390x and ppcle64. https://github.com/confidential-containers/guest-components/issues/422

Reply:
Good point, I was wondering how soon it would be before I hit that. When I checked last week there still hadn't been a release of `tough` since Ding's fix was merged :(

Reply:
Hi @ChengyuZhu6 @stevenhorsman ,

This assert seems to still fail occasionally and I don´t know why. For example, on SNP job of today's update:
[kata-containers-ci-on-push / run-kata-coco-tests / run-k8s-tests-sev-snp (qemu-snp, nydus, guest-pull)](https://github.com/kata-containers/kata-containers/actions/runs/8262539951/job/22611212779?pr=8484#logs)

Same is passing on SEV: [kata-containers-ci-on-push / run-kata-coco-tests / run-k8s-tests-on-sev (qemu-sev, nydus, guest-pull)](https://github.com/kata-containers/kata-containers/actions/runs/8262539951/job/22611213394?pr=8484#logs)

I've a comment for the `sandbox_id` right below that might be an issue to consider.

Reply:
Caught my attention on the snp job mentioned above the following message:

```
# sandbox_id is: f7b8582f849754b062618f5d7f805fcee0c7f34c7dc5b354cd7025a44f9e59ec 51e4bf885526dcc9f89559009e845df1bea76189523018b925f4f7029287c42b 25b4b2f08f6ad3c17d5c5f42f78d6b8823b2b902edf51c9e069df5839f6f6de6 da28503a567a862e1eb2e326c9432198c009d1ff6633d4492c85fd20cfd2081b
```

It seems that `get_node_kata_sandbox_id()` returned several IDs, because likely there are dangling pods on the system. That function should return only one ID accurately.

Reply:
We have some tests that starts by cleaning up the environment to ensure no dangling pods/processes/shims are running. We might need to implement that here too

Reply:
> Caught my attention on the snp job mentioned above the following message:
> 
> ```
> # sandbox_id is: f7b8582f849754b062618f5d7f805fcee0c7f34c7dc5b354cd7025a44f9e59ec 51e4bf885526dcc9f89559009e845df1bea76189523018b925f4f7029287c42b 25b4b2f08f6ad3c17d5c5f42f78d6b8823b2b902edf51c9e069df5839f6f6de6 da28503a567a862e1eb2e326c9432198c009d1ff6633d4492c85fd20cfd2081b
> ```
> 
> It seems that `get_node_kata_sandbox_id()` returned several IDs, because likely there are dangling pods on the system. That function should return only one ID accurately.

@wainersm I’ve also observed this. Do we need cleanup ci environment on snp?

Reply:
@ChengyuZhu6 I think so, yes. However, even if we manually clean up manually it now then later this problem may come back to bite us :D 

Ideally the tests should ensure the system is in a good state and do the clean ups if needed.

I will look for examples of tests that do the clean up and we can adapt to this.